koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -369,8 +356,35 @@ def _get_error_message(self, response):
 
         return error_message
 
-    def handler(self, src, target_info, arch, scratch=False,
-                yum_repourls=None, branch=None, push_url=None):
+    def check_whitelist(self, name, target_info):
+        """Check if container name is whitelisted in destination tag
+
+        Raises with koji.BuildError if package is not whitelisted or blocked.
+        """
+        pkg_cfg = self.session.getPackageConfig(target_info['dest_tag_name'],
+                                                name)
+        self.logger.debug("%r" % pkg_cfg)
+        # Make sure package is on the list for this tag
+        if pkg_cfg is None:
+            raise koji.BuildError("package (container) %s not in list for tag %s" % (name, target_info['dest_tag_name']))
+        elif pkg_cfg['blocked']:
+            raise koji.BuildError("package (container)  %s is blocked for tag %s" % (name, target_info['dest_tag_name']))
+
+    def runBuilds(self, src, target_info, arches, scratch=False,
+                  yum_repourls=None, branch=None, push_url=None):
+
+        self.logger.debug("Spawning jobs for arches: %r" % (arches))
+
+        results = []
+        for arch in arches:
+            result = self.createContainer(
```

Comment:
Right, this is a temporary solution. Ultimately osbs-client will take care of running arch-specific builds and assembling the builds, but since its not yet implemented this part will be left as is